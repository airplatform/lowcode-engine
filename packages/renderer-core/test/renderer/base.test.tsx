import renderer from 'react-test-renderer';
import React from 'react';
import '../utils/react-env-init';
import pageRendererFactory from '../../src/renderer/renderer';
import { sampleSchema } from '../mock/sample';
import loopSchema from '../mock/loop';

describe('notFountComponent', () => {
  const Render = pageRendererFactory();

  const component = renderer.create(
    // @ts-ignore
    <Render
      schema={sampleSchema as any}
      components={{}}
      appHelper={{}}
    />,
  );

  it('not found snapshot', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('loop schema', () => {
  it('loop key', () => {
    const Render = pageRendererFactory();

    const component = renderer.create(
      // @ts-ignore
      <Render
        schema={loopSchema as any}
        components={{}}
        appHelper={{}}
      />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})