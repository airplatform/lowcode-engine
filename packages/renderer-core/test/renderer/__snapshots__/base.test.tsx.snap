// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loop schema loop key 1`] = `
<div
  className="lce-page _css_pseudo_node_ocl1djd9o41"
  style={Object {}}
>
  <div
    __componentName="RootHeader"
    __id="node_ocl1djd9o42"
  >
    Component Not Found
  </div>
  <div
    __componentName="RootContent"
    __id="node_ocl1djd9o43"
    contentBgColor="white"
    contentMargin="20"
    contentPadding="20"
  >
    <div
      __componentName="Div"
      __id="node_ocl1djd9o45"
      __style__={Object {}}
      behavior="NORMAL"
      className="_css_pseudo_node_ocl1djd9o45"
      customClassName=""
      events={
        Object {
          "ignored": true,
        }
      }
      fieldId="div_l1djdj1n"
      useFieldIdAsDomId={false}
    >
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
    </div>
    <div
      __componentName="Div"
      __id="node_ocl1djd9o45"
      __style__={Object {}}
      behavior="NORMAL"
      className="_css_pseudo_node_ocl1djd9o45"
      customClassName=""
      events={
        Object {
          "ignored": true,
        }
      }
      fieldId="div_l1djdj1n"
      useFieldIdAsDomId={false}
    >
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
    </div>
    <div
      __componentName="Div"
      __id="node_ocl1djd9o45"
      __style__={Object {}}
      behavior="NORMAL"
      className="_css_pseudo_node_ocl1djd9o45"
      customClassName=""
      events={
        Object {
          "ignored": true,
        }
      }
      fieldId="div_l1djdj1n"
      useFieldIdAsDomId={false}
    >
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
      <div
        __componentName="Div"
        __id="node_ocl1djd9o46"
        __style__={Object {}}
        behavior="NORMAL"
        className="_css_pseudo_node_ocl1djd9o46"
        customClassName=""
        events={
          Object {
            "ignored": true,
          }
        }
        fieldId="div_l1djdj1o"
        useFieldIdAsDomId={false}
      >
        Component Not Found
      </div>
    </div>
  </div>
  <div
    __componentName="RootFooter"
    __id="node_ocl1djd9o44"
  >
    Component Not Found
  </div>
</div>
`;

exports[`notFountComponent not found snapshot 1`] = `
<div
  className="lce-page _css_pseudo_node_ockyigdqxl1"
  style={Object {}}
>
  <div
    __componentName="RootHeader"
    __id="node_ockyigdqxl2"
  >
    Component Not Found
  </div>
  <div
    __componentName="RootContent"
    __id="node_ockyigdqxl3"
    contentBgColor="white"
    contentMargin="20"
    contentPadding="20"
  >
    <div
      __componentName="Button"
      __id="node_ockyigdqxl5"
      __style__={Object {}}
      baseIcon=""
      behavior="NORMAL"
      className="_css_pseudo_node_ockyigdqxl5"
      content="按 钮"
      events={
        Object {
          "ignored": true,
        }
      }
      fieldId="button_kyige3yf"
      loading={false}
      otherIcon=""
      size="medium"
      triggerEventsWhenLoading={false}
      type="primary"
    >
      Component Not Found
    </div>
  </div>
  <div
    __componentName="RootFooter"
    __id="node_ockyigdqxl4"
  >
    Component Not Found
  </div>
</div>
`;
